src/common/Utilities/SFMTRand.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2008-2019 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SFMTRand_h__
+#define SFMTRand_h__
+
+#include <SFMT.h>
+
+namespace boost {
+    template <typename T> class thread_specific_ptr;
+}
+
+/*
+ * C++ Wrapper for SFMT
+ */
+class SFMTRand {
+    friend class boost::thread_specific_ptr<SFMTRand>;
```

Comment:
When the constructor is public, there is no need for friend declarations

Reply:
removed